libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -148,12 +156,16 @@ void CodeTransform::freeUnusedVariables(bool _popUnusedSlotsAtStackTop)
 		return;
 
 	for (auto const& identifier: m_scope->identifiers)
-		if (holds_alternative<Scope::Variable>(identifier.second))
-		{
-			Scope::Variable const& var = std::get<Scope::Variable>(identifier.second);
-			if (m_variablesScheduledForDeletion.count(&var))
-				deleteVariable(var);
-		}
+		if (Scope::Variable const* var = get_if<Scope::Variable>(&identifier.second))
+			if (m_variablesScheduledForDeletion.count(var))
+				deleteVariable(*var);
+	// Directly in a function body block, we can also delete the function arguments,
```

Comment:
Is this safe for nested functions like
```
function f(a) { function g() {} }
```?

Reply:
I'll add ``!m_scope->functionScope``, then it should be perfectly safe.